es6-shim.js

```diff
@@ -2842,7 +2842,11 @@
         };
 
         MapIterator.prototype = {
+          isMapIterator: true,
```

Comment:
it occurs to me that this is too easily forgeable; but that since `instanceof` can't work cross-realm, this might be the best we can get.